### PR TITLE
Access control: Reduce number of API calls for role picker

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
Due to the way the state of UserRolePicker and TeamRolePicker was structured, roles for users and teams was fetched several times. four times for each user and two times for each team.

By moving the state of roleOptions and builtInRoles up one component we can get the expected behaviour of 1 call for each user/team

### CODE REVIEW

File: public/app/features/admin/UserOrgs.tsx [hunk 1]
```diff
@@ -133,8 +134,21 @@ class UnThemedOrgRow extends PureComponent<OrgRowProps> {
...
     currentRole: this.props.org.role,
     isChangingRole: false,
+    roleOptions: [],
+    builtInRoles: {},
   };
 
+  componentDidMount() {
+    if (contextSrv.licensedAccessControlEnabled()) {
+      if (contextSrv.hasPermission(AccessControlAction.ActionRolesList)) {
+        fetchRoleOptions(this.props.org.orgId).then((roles) => this.setState({ roleOptions: roles }));
```
Comment (Clarity-89) [inline public/app/features/admin/UserOrgs.tsx:144]:
```suggestion
        fetchRoleOptions(this.props.org.orgId).then((roles) => this.setState({ roleOptions: roles })).catch(err => console.error(err));
```
Should probably handle an error here too.

File: public/app/features/admin/UserOrgs.tsx [hunk 2]
```diff
@@ -133,8 +134,21 @@ class UnThemedOrgRow extends PureComponent<OrgRowProps> {
...
+      }
+      if (contextSrv.hasPermission(AccessControlAction.ActionBuiltinRolesList)) {
+        fetchRoleOptions(this.props.org.orgId).then((roles) => this.setState({ builtInRoles: roles }));
```
Comment (Clarity-89) [inline public/app/features/admin/UserOrgs.tsx:147]:
```suggestion
        fetchRoleOptions(this.props.org.orgId).then((roles) => this.setState({ builtInRoles: roles })).catch(err => console.error(err));
```

File: public/app/features/teams/TeamList.tsx [hunk 3]
```diff
@@ -65,6 +65,10 @@ export class TeamList extends PureComponent<Props, State> {
     this.props.setSearchQuery(value);
   };
 
+  getRoleOptions = () => {
```
Comment (Clarity-89) [inline public/app/features/teams/TeamList.tsx:68]: This doesn't seem to be used anywhere? Also seems weird to wrap state in promise just to make a return value async.

**Review by Clarity-89 — APPROVED** [review]
Great work 🎉   Tested and works great, also the weird flickering is gone. Left a few minor comments.

File: public/app/features/admin/UserOrgs.tsx (hunk 1 above)
Comment (kalleep) [reply public/app/features/admin/UserOrgs.tsx:144]: Yes!

File: public/app/features/teams/TeamList.tsx (hunk 3 above)
Comment (kalleep) [reply public/app/features/teams/TeamList.tsx:68]: Yes i agree, and it was how it worked before. This was added by mistake. I added it to test if the issue was anonymous functions triggering all the extra api calls which it was

**Pull request merged**